R/brreg.R

```diff
@@ -0,0 +1,76 @@
+#' Get entity
+#'
+#' Collect basic information about an entity from the \emph{Central Coordinating
+#' Register for Legal Entities} (Enhetsregisteret) based on the entity's
+#' organisation number.
+#'
+#' See the API documentation for further details
+#' \url{https://data.brreg.no/enhetsregisteret/api/docs/index.html} (in
+#' Norwegian only).
+#'
+#' @param entity A Norwegian organisation number or company name.
+#' @param raw_response  If TRUE a list of class `noAPI` is
+#'   returned, including the raw `httr2_response`.
+#' @return data.frame or list
+#' @export
+#' @examples
+#' # Get entity by number
+#' df <- get_entity(974760843)
+#' # Get entity by name
+#' df <- get_entity('Riksrevisjonen')
+#' # Get multiple entites
+#' df <- get_entity(c(974760843, 971524960))
+#'
+get_entity <- function(entity, raw_response = FALSE) {
+  dl <- lapply(entity, function(x) get_entity_safe(x, raw_response))
+  if (raw_response) return(dl)
+  do.call('rbind', dl)
+}
+
+#' Get municipalities
+#'
+#' Collect the names and keys for Norwegian municipalities from the
+#' \emph{Central Coordinating Register for Legal Entities} (Enhetsregisteret).
+#'
+#' @inheritParams get_entity
+#' @return data.frame or list
+#' @export
+#' @examples
+#' # Get municipalities
+#' df <- get_municipalities()
+#'
+get_municipalities <- function(raw_response = FALSE) {
+
+  resp <- request_brreg('kommuner')
+  parsed <- resp_body_json(resp, simplifyVector = TRUE)
+
+  if (raw_response) {
+    out <- make_api_object(resp, parsed)
+  }
+  else {
+    out <- parsed$`_embedded`$kommuner[,1:2]
+  }
+  return(out)
+
+}
+
+#' Get entity (single method)
+#' @inheritParams get_entity
+#' @noRd
+get_entity_ <- function(entity, raw_response = FALSE) {
+  resp <- request_brreg('enheter', entity)
+  parsed <- parse_response(resp, simplifyVector = FALSE)
+  if (raw_response) {
```

Comment:
Use oneliner?

```r
out <- if (raw_response) make_api_object(resp, parsed) else parse_brreg_entity(parsed)
```

Reply:
I'm sort of leaning towards explict `{}` statements, as this (I think) increasingly is becoming the standard convention.

Reply:
I'm not sure if I'm totally into the extremely verbose coding style in Hadleyverse, and I sort of always want to rebel against conventions as well. On the other side, `<- if (cond) ... else ...` only works with short lines, and this line will become a tad too long for my liking, so `{}` might be the best choice anyway.

Reply:
The better argument is obviously that the ternary variant is more performant than the verbose one, albeit only slightly:

```r
funA <- \(x) {
  if (x == 'A') {
    x <- 0
  } else {
    x <- 1
  }
  return(x)
}

funB <- \(x) if (x == 'A') 0 else 1

microbenchmark::microbenchmark(funA('A'), funB('A'), times = 10000)
```

Reply:
Yep. Full agree on one-liner `if {} else {}` for shorter lines being acceptable, but I do think this one is a bit long.

Reply:
> The better argument is obviously that the ternary variant is more performant than the verbose one, albeit only slightly:
> 
> ```r
> funA <- \(x) {
>   if (x == 'A') {
>     x <- 0
>   } else {
>     x <- 1
>   }
>   return(x)
> }
> 
> funB <- \(x) if (x == 'A') 0 else 1
> 
> microbenchmark::microbenchmark(funA('A'), funB('A'), times = 10000)
> ```

Interesting. Also good point.